library/ssl_cli.c

```diff
@@ -3206,8 +3212,10 @@ static int ssl_parse_certificate_request( mbedtls_ssl_context *ssl )
 #else /* MBEDTLS_KEY_EXCHANGE_CERT_REQ_ALLOWED_ENABLED */
 static int ssl_parse_certificate_request( mbedtls_ssl_context *ssl )
 {
+
+    const unsigned char *hs_msg;
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
-    unsigned char *buf;
+    const unsigned char *buf;
```

Comment:
General note: It would ease review if any changes different from plain replacement of `ssl->in_msg` by `mbedtls_ssl_hs_msg_ptr( ssl )` were handled in separate commits.

Reply:
Thanks, I will split the commits.

Reply:
Please see these commits:

03f1e7b - handshake message accessor (equivalent to `ssl->in_msg`)
5c4321  - handshake message length (equivalent to `ssl->in_hslen`)
b49671 - handshake message body length (equivalent to `ssl->in_hslen - mbedtls_ssl_hs_hdr_len( ssl )`)
12ef5a9 - handshake message type (equivalent to `ssl->in_msg[0]`)